src/pages/billing/Project.js

```diff
@@ -26,16 +28,21 @@ import { billingRoles } from 'src/pages/billing/List'
 const workspaceLastModifiedWidth = 150
 const workspaceExpandIconSize = 20
 
-const workspaceBillingStatusIconOrEmpty = (() => {
+const BillingAccountIcon = {
+  updating: { shape: 'sync', color: 'orange' },
+  done: { shape: 'check', color: colors.accent() },
+  error: { shape: 'warning-standard', color: 'red' }
+}
+
+const getBillingAccountIconOrEmpty = (() => {
```

Comment:
The short circuiting could look like a line of:
`if(!icon) return blank`
and then `const { shape, color } = icon` and then the `||` at the end goes away.

Reply:
My idea was that `getBillingAccountIconOrEmpty` would have the following signature: 
```haskell
getBillingAccountIconOrEmpty :: Maybe BillingAccountStatus -> Icon
```

Reply:
So, if you do the destructuring in the signature, the problem you get is that the interpreter is going to try to reach into a non-existent object when it is undefined. What you want is something like this:
```
const fn = (icon) => {
 if (icon === undefined) {
  console.log('undefined')
  return;
 }
 const {foo, bar} = icon
 console.log(`${foo}/${bar}`)
}
```
What that will do is give you a function that logs ${foo}/${bar} when icon is defined, and logs 'undefined' if you call it as `fn()`.

Reply:
The reason you got the compiler error, which I am assuming was something like: `Uncaught TypeError: Cannot destructure property 'foo' of 'undefined' as it is undefined.` was that javascript was protecting you against a null pointer if you called it with no arg. Because otherwise it will try to index into undefined, which wouldn't work.

(As a side note, it will be undefined as the above, not null. They're actually two different constructs in JS.)

Reply:
Thanks for the explanation. I assumed that something like this was happening. I wish it didn't do this though. JS bites your head off in so many ways, it feels odd that it would protect you here, expecially when binding `null` to an object is a reasonable thing to do IMO.